nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchRecord.java

```diff
@@ -0,0 +1,355 @@
+public class PutElasticsearchRecord extends AbstractProcessor implements ElasticsearchRestProcessor {
+    static final PropertyDescriptor RECORD_READER = new PropertyDescriptor.Builder()
+        .name("put-es-record-reader")
+        .displayName("Record Reader")
+        .description("The record reader to use for reading incoming records from flowfiles.")
+        .identifiesControllerService(RecordReaderFactory.class)
+        .required(true)
+        .build();
+
+    static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+        .name("put-es-record-batch-size")
+        .displayName("Batch Size")
+        .description("The number of records to send over in a single batch.")
+        .defaultValue("100")
+        .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .required(true)
+        .build();
+
+    static final PropertyDescriptor ID_RECORD_PATH = new PropertyDescriptor.Builder()
+        .name("put-es-record-id-path")
+        .displayName("ID Record Path")
+        .description("A record path expression to retrieve the ID field for use with Elasticsearch. If left blank " +
+                "the ID will be automatically generated by Elasticsearch.")
+        .addValidator(new RecordPathValidator())
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .build();
+
+    static final PropertyDescriptor INDEX_RECORD_PATH = new PropertyDescriptor.Builder()
+        .name("put-es-record-index-record-path")
+        .displayName("Index Record Path")
+        .description("A record path expression to retrieve the index field for use with Elasticsearch. If left blank " +
+                "the index will be determined using the main index property.")
+        .addValidator(new RecordPathValidator())
+        .required(false)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .build();
+
+    static final PropertyDescriptor TYPE_RECORD_PATH = new PropertyDescriptor.Builder()
+        .name("put-es-record-type-record-path")
+        .displayName("Type Record Path")
+        .description("A record path expression to retrieve the type field for use with Elasticsearch. If left blank " +
+                "the type will be determined using the main type property.")
+        .addValidator(new RecordPathValidator())
+        .required(false)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .build();
+
+    static final PropertyDescriptor ERROR_RECORD_WRITER = new PropertyDescriptor.Builder()
+        .name("put-es-record-error-writer")
+        .displayName("Error Record Writer")
+        .displayName("If this configuration property is set, the response from Elasticsearch will be examined for failed records " +
```

Comment:
This should be `description` rather than a duplicate `displayName`, I'll change it locally to continue testing and can change it before merge unless you'd like to